opencv_ai/splice/component.h

```diff
@@ -0,0 +1,60 @@
+#ifndef OPENCV_AI_SPLICE_COMPONENT_H_
+#define OPENCV_AI_SPLICE_COMPONENT_H_
+
+#include <boost/optional.hpp>
+#include <boost/regex.hpp>
+
+#include <glog/logging.h>
+
+#include <opencv_ai/splice/context.h>
+
+namespace opencv_ai {
```

Comment:
Well.. thinking that the toplevel company has a namespace of ``opencv_ai``, which matches the toplevel include directory, and sub-packages live in a sub namespace which matches their include directory, such as ``slice``, if it were available(doh), or we could have a single company wide namespace, and not have a sub namespace per package.  I prefer the single namespace because I have so rarely run out of names working on a project mostly contributed to by only one company... but could be sold on having a namespace per package (more typing...).